packages/react-avatar/src/components/Avatar/useAvatarStyles.ts

```diff
@@ -0,0 +1,319 @@
+
+const animationDuration = {
+  duration50: '50ms',
+  duration100: '100ms',
+  duration150: '150ms',
+  duration200: '200ms',
+  duration300: '300ms',
+  duration400: '400ms',
+  duration500: '500ms',
+};
+
+const animationTiming = {
+  ultraFast: animationDuration.duration50,
+  faster: animationDuration.duration100,
+  fast: animationDuration.duration150,
+  normal: animationDuration.duration200,
+  slow: animationDuration.duration300,
+  slower: animationDuration.duration400,
+  ultraSlow: animationDuration.duration500,
+};
+
+const animationLines = {
+  decelerateMax: 'cubic-bezier(0.00,0.00,0.00,1.00)',
+  decelerateMid: 'cubic-bezier(0.10,0.90,0.20,1.00)',
+  decelerateMin: 'cubic-bezier(0.33,0.00,0.10,1.00)',
+  accelerateMax: 'cubic-bezier(1.00,0.00,1.00,1.00)',
+  accelerateMid: 'cubic-bezier(0.90,0.10,1.00,0.20)',
+  accelerateMin: 'cubic-bezier(0.80,0.00,0.78,1.00)',
+  maxEasyEase: 'cubic-bezier(0.80,0.00,0.20,1.00)',
+  easyEase: 'cubic-bezier(0.33,0.00,0.67,1.00)',
+  linear: 'linear',
+};
+
+const animations = {
+  fastOutSlowInMax: animationLines.decelerateMax,
+  fastOutSlowInMid: animationLines.decelerateMid,
+  fastOutSlowInMin: animationLines.decelerateMin,
+  slowOutFastInMax: animationLines.accelerateMax,
+  slowOutFastInMid: animationLines.accelerateMid,
+  slowOutFastInMin: animationLines.accelerateMin,
+  fastEase: animationLines.maxEasyEase,
+  normalEase: animationLines.easyEase,
+  nullEasing: animationLines.linear,
+};
+
+const iconSize = {
+  small: '12px',
+  medium: '16px',
+  large: '24px',
+  larger: '40px',
+  largest: '48px',
+};
+
+const activeRing = (s: AvatarState) =>
+  (s.active === 'active' || s.active === 'inactive') &&
+  (!s.activeDisplay ||
+    s.activeDisplay === 'ring' ||
+    s.activeDisplay === 'ring-glow' ||
+    s.activeDisplay === 'ring-shadow');
+
+const activeShadow = (s: AvatarState) =>
+  (s.active === 'active' || s.active === 'inactive') &&
+  (s.activeDisplay === 'shadow' || s.activeDisplay === 'ring-shadow');
+
+const activeGlow = (s: AvatarState) =>
+  (s.active === 'active' || s.active === 'inactive') && (s.activeDisplay === 'glow' || s.activeDisplay === 'ring-glow');
+
+const useStyles = makeStyles<AvatarState>([
+  [
+    null,
+    tokens => ({
+      display: 'inline-block',
+      flexShrink: 0,
+      position: 'relative',
+      verticalAlign: 'middle',
+
+      borderRadius: tokens.global.borderRadius.circular,
+
+      width: `${defaultAvatarSize}px`,
+      height: `${defaultAvatarSize}px`,
+
+      fontFamily: tokens.global.type.fontFamilies.base,
+      fontSize: tokens.global.type.fontSizes.base[300],
+      fontWeight: tokens.global.type.fontWeights.semibold,
+    }),
+  ],
+
+  // Add rules to set the width and height based on every possible size
+  ...avatarSizeValues.map(
+    val => [s => s.size === val, { width: `${val}px`, height: `${val}px` }] as [(s: AvatarState) => boolean, {}],
+  ),
```

Comment:
@behowell FYI: I am not sure that will be able to support such definitions via build transforms...